### PR TITLE
Implement account, session and utility sizes

### PR DESCRIPTION
**Pull request by niklaslong**
Dipping my toes into C NIFs—wasn't sure of the type to use but figured `enif_make_ulong` was probably close to what we need?

### CODE REVIEW

File: lib/olm.ex [hunk 1]
```diff
@@ -7,7 +7,25 @@ defmodule Olm do
...
+
+  @doc """
+  The size of an session object in bytes.
+  """
+  def session_size(), do: error("session_size/0")
+
+  @doc """
+  The size of an utiliy object in bytes.
+  """
+  def utility_size(), do: error("utiliy_size/0")
```
Comment (fancycade) [inline lib/olm.ex]: Misspelled utility in the error message

File: native/src/olm_nifs.c [hunk 2]
```diff
@@ -17,10 +16,34 @@ version(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
 			    enif_make_int(env, patch));
 }
 
+static ERL_NIF_TERM
+account_size(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
+    size_t size = olm_account_size();
+	
+	return enif_make_ulong(env, size);
```
Comment (fancycade) [inline native/src/olm_nifs.c]: Spacing isn't aligned.

File: lib/olm.ex [hunk 3]
```diff
@@ -7,7 +7,25 @@ defmodule Olm do
...
+  """
+  def account_size(), do: error("account_size/0")
+
+  @doc """
+  The size of an session object in bytes.
+  """
+  def session_size(), do: error("session_size/0")
+
+  @doc """
+  The size of an utiliy object in bytes.
```
Comment (fancycade) [inline lib/olm.ex]: Misspelled utility in doc

**Review by fancycade — CHANGES_REQUESTED** [review]
Good start! Had to do some research on the int typing. Looks like ulong will be the most generic so that is fine.

We should probably make version use enif_make_uint instead of enif_make_int now that I think about it. I didn't look far enough in the erlang docs :sweat_smile:

File: lib/olm.ex (hunk 1 above)
Comment (niklaslong) [reply lib/olm.ex]: Good eye! 👀

**Comment by niklaslong** [comment]
@fancycade I've added a simple clang-format, let me know if you have particular preferences you think should be included. I've mostly based it on Webkit and Mozilla styles with 4 char indent à la Rust. Here's a good tool to play around with configuration: https://zed0.co.uk/clang-format-configurator/

Formatting Elixir + C files can be done with `mix fmt`.

**Review by fancycade — APPROVED** [review]

**Pull request merged**